vcd/resource_vcd_catalog_vapp_template_test.go

```diff
@@ -0,0 +1,321 @@
+//go:build catalog || ALL || functional
+// +build catalog ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccVcdCatalogVAppTemplateResource(t *testing.T) {
+	vAppTemplateName := t.Name()
+	vAppTemplateDescription := vAppTemplateName + "Description"
+	vAppTemplateFromUrlName := t.Name() + "FromUrl"
+
+	preTestChecks(t)
+
+	if testConfig.Ova.OvfUrl == "" {
+		t.Skip("Variable Ova.OvfUrl must be set in test configuration")
+	}
+
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Vdc":              testConfig.VCD.Vdc, // TODO: Use NSX-T VDC by default
```

Comment:
In fact - would be good if you could switch it. This gives us two things:
* Gets us ready for NSX-V deprecation
* Covers CDS tests

Reply:
Yep, you're right. I noted that `TODO` there to do it in the following PR (the one that adjusts `vcd_vm` and `vcd_vapp_vm`) as this PR was quite big and confusing as it is now.

The reason is that, if I remember correctly, this test uses some pre-created catalog and vApp Template that is created by a shared function before the test starts. Fearing that this function is used in more places (it is in `config_test.go`), and not touch more things in this PR, I just left the comment to not forget.